components/ApplyToHostModal.js

```diff
@@ -341,7 +341,7 @@ const ApplyToHostModal = ({ hostSlug, collective, onClose, onSuccess, router, ..
                               creatable
                               renderNewCollectiveOption={() => (
                                 <Link
-                                  href={isOCFHost ? `/foundation/apply/intro` : `/${host.slug}/apply`}
+                                  href={isOCFHost ? `/foundation/apply/intro` : `/${host.slug}/apply-new`}
```

Comment:
> The Foundation application process (with "Create new collective") is not going to the custom flow. Going to `/foundation/apply-new` instead of `https://opencollective.com/foundation/apply/intro`

This should be working as per logic here. `isOCFHost` is being calculated via `host.legacyId` which is specific for [`production` and `staging`](https://github.com/opencollective/opencollective-frontend/search?q=OPENCOLLECTIVE_FOUNDATION_ID) environment so this might not work in local env unless we use our local db id in the constant.